### PR TITLE
jupyter web app manifest version v3

### PR DESCRIPTION
**Pull request by jlewi**
* Define v3 versions of the manifests that use the patterns as described
in http://bit.ly/kf_kustomize_v3

* Define a new base_v3 package
* Use a configmap generator to generate a config map containing application specific parameters
* Make the parameters available to the command line by setting environment variables to
the config map and then referencing them in the command line.

Related to #774

### CODE REVIEW

**Comment by jlewi** [comment]
Test failure looks like a git flake.
#1013 should probably be merged first.

**Comment by jlewi** [comment]
build and deploy failed.

```
util.py                     72 INFO     time="2020-03-30T23:39:03Z" level=error msg="Permanently failed applying application jupyter-web-app; error:  (kubeflow.error): Code 500 with message: Apply.Run  Error error when creating \"/tmp/kout192295368\":
admission webhook \"pilot.validation.istio.io\" denied the request: configuration is invalid: domain name \"kubeflow.svc.$(clusterDomain)\" invalid (label \"$(clusterDomain)\" invalid)" filename="kustomize/kustomize.go:247"
util.py                     72 INFO     Error: failed to apply:  (kubeflow.error): Code 500 with message: kfApp Apply failed for kustomize:  (kubeflow.error): Code 500 with message: Apply.Run  Error error when creating "/tmp/kout192295368": admission w
ebhook "pilot.validation.istio.io" denied the request: configuration is invalid: domain name "kubeflow.svc.$(clusterDomain)" invalid (label "$(clusterDomain)" invalid)
```

**Comment by jlewi** [comment]
@kimwnasptd The tests passed could you please review.

File: jupyter/jupyter-web-app/base/configs/spawner_ui_config.yaml [hunk 1]
```diff
@@ -18,7 +18,11 @@ spawnerFormDefaults:
   image:
     # The container Image for the user's Jupyter Notebook
     # If readonly, this value must be a member of the list below
+<<<<<<< HEAD
```
Comment (kimwnasptd) [inline jupyter/jupyter-web-app/base/configs/spawner_ui_config.yaml:21]: I think we should go with the HEAD changes :)
Also the generated test data has these lines, I guess you'll have to regenerate the tests?

Comment (jlewi) [reply jupyter/jupyter-web-app/base/configs/spawner_ui_config.yaml]: Good catch thanks.

**Pull request merged**